src/main/java/org/springframework/hateoas/client/Hop.java

```diff
@@ -0,0 +1,50 @@
+package org.springframework.hateoas.client;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Container for cuztomizations to a single traverson "hop"
+ *
+ * @author Greg Turnquist
+ */
+public class Hop {
```

Comment:
Let's make `Hop` a value object with `with…(…)` methods for chaining. Having `with…` alongside `set…` seems weird. Getters can be package protected I guess. Make sure the methods have JavaDoc. Lomboks' `@Wither` and `@Value` could help reducing implementation effort.